core/integration/bookkeeping/client.go

```diff
@@ -0,0 +1,140 @@
+/*
+ * === This file is part of ALICE O² ===
+ *
+ * Copyright 2022 CERN and copyright holders of ALICE O².
+ * Author: Claire Guyot <claire.guyot@cern.ch>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * In applying this license CERN does not waive the privileges and
+ * immunities granted to it by virtue of its status as an
+ * Intergovernmental Organization or submit itself to any jurisdiction.
+ */
+
+package bookkeeping
+
+import (
+	"context"
+	"time"
+
+	"github.com/AliceO2Group/Control/common/logger"
+	bkpb "github.com/AliceO2Group/Control/core/integration/bookkeeping/protos"
+	//bkpb "github.com/AliceO2Group/Bookkeeping"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/viper"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/backoff"
+	"google.golang.org/grpc/connectivity"
+	"google.golang.org/grpc/keepalive"
+)
+
+var log = logger.New(logrus.StandardLogger(), "bookkeepingclient")
+
+type RpcClient struct {
+	bkpb.EnvironmentServiceClient
+	bkpb.FlpServiceClient
+	bkpb.LogServiceClient
+	bkpb.RunServiceClient
```

Comment:
It's not a problem, but you should be aware of how method redirection behaves in case of naming collision so you don't end up with behaviour you didn't expect. Basically you'll have to call them explicitly through one of the composed service objects rather than just `client.Create` (this is an odd thing to do with composition because it defeats some of the point of it, but I believe you're already doing it in all instances, effectively using composed services as if they were member variables, so in this case we should be ok).

See https://genekuo.medium.com/composition-with-structures-and-method-forwarding-in-go-9a9045f2c1fd the section **Naming Collisions**.